ocs_ci/templates/mcg/java/s3test/src/main/java/amazons3/s3test/ChunkedUploadApplication.java

```diff
@@ -0,0 +1,231 @@
+				.withPayloadSigningEnabled(true).withClientConfiguration(clientConfiguration).build();
+
+		return s3Client;
+	}
+
+	// regularUpload uploads file to bucket using s3Client
+	public static Boolean regularUpload(AmazonS3 s3Client, String bucketName, String KEY, File originalFile) {
+		try {
+			PutObjectRequest request = new PutObjectRequest(bucketName, KEY, originalFile);// new File(PATH));
+			s3Client.putObject(request);
+			return true;
+		} catch (AmazonServiceException e) {
+			e.printStackTrace();
+		} catch (SdkClientException e) {
+			e.printStackTrace();
+		}
+		return false;
+	}
+
+	// multiPartUpload uploads file to bucket using s3Client using multipartUpload
+	public static Boolean multiPartUpload(AmazonS3 s3Client, String bucketName, String KEY, File originalFile) {
+		// File file = new File(PATH);
+		long contentLength = originalFile.length();
+		long partSize = 5 * 1024 * 1024; // Set part size to 5 MB.
+		try {
+			List<PartETag> partETags = new ArrayList<PartETag>();
+
+			// init request
+			InitiateMultipartUploadRequest initRequest = new InitiateMultipartUploadRequest(bucketName, KEY);
+			InitiateMultipartUploadResult initResponse = s3Client.initiateMultipartUpload(initRequest);
+
+			// Upload the file parts
+			long filePosition = 0;
+			for (int i = 1; filePosition < contentLength; i++) {
+				partSize = Math.min(partSize, (contentLength - filePosition)); // last part <= 5MB
+
+				UploadPartRequest uploadRequest = new UploadPartRequest().withBucketName(bucketName).withKey(KEY)
+						.withUploadId(initResponse.getUploadId()).withPartNumber(i).withFileOffset(filePosition)
+						.withFile(originalFile).withPartSize(partSize);
+
+				UploadPartResult uploadResult = s3Client.uploadPart(uploadRequest);
+				partETags.add(uploadResult.getPartETag());
+				filePosition += partSize;
+			}
+
+			// Complete the multipart upload.
+			CompleteMultipartUploadRequest compRequest = new CompleteMultipartUploadRequest(bucketName, KEY,
+					initResponse.getUploadId(), partETags);
+			s3Client.completeMultipartUpload(compRequest);
+		} catch (AmazonServiceException e) {
+			e.printStackTrace();
+			return false;
+		} catch (SdkClientException e) {
+
+			e.printStackTrace();
+			return false;
+		}
+		return true;
+	}
+
+	// download download file from bucket using s3Client
+	public static Boolean download(AmazonS3 s3Client, String bucketName, String KEY, File localFile)
+			throws IOException {
+		try {
+			GetObjectRequest request = new GetObjectRequest(bucketName, KEY);
+			s3Client.getObject(request, localFile);
+		} catch (AmazonServiceException e) {
+			e.printStackTrace();
+			return false;
+		} catch (SdkClientException e) {
+			e.printStackTrace();
+			return false;
+		}
+		return true;
+	}
+
+	// compareMD5 compares MD5s of file downloaded from bucket using s3Client and
+	// original file
+	public static boolean compareMD5(String PATH1, String PATH2) throws IOException, NoSuchAlgorithmException {
```

Comment:
This approach for comparison (byte array) is a bit expensive in terms of memory usage since you read all the file (2 files in this case) into the memory and compare them.
Here you're dealing with 5MiB files but if you set larger files in size, it could be problematic.
Please check and consider using BufferedInputStreams instead of Files.readAllBytes